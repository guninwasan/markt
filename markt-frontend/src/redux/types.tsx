type UserAuthType = {
  userID: number;
  name: string;
  email: string;
  phone: string;
  jwt: string;
};

type DeviceSliceType = {
  isLoggedIn: boolean;
  isLoading: boolean;
};

export type { UserAuthType, DeviceSliceType };
