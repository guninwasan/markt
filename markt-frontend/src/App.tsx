import React from "react";
import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
import { ProductListingPage } from "./pages";
import { useIsMobile } from "./hooks";
import { HomePage } from "./pages/home-page";

function App() {
  useIsMobile();
  return (
    <Router>
      <Routes>
        <Route path="/" element={<HomePage />} />
        <Route path="/listing" element={<ProductListingPage />} />
        <Route path="*" element={<h1>Not Found</h1>} />
      </Routes>
    </Router>
  );
}

export default App;
