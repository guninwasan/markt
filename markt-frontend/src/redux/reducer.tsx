import { combineReducers } from "@reduxjs/toolkit";
import userAuth from "./slices/user-auth-slice";
import device from "./slices/device-slice";

const reducer = combineReducers({
  userAuth,
  device,
});

export { reducer };
