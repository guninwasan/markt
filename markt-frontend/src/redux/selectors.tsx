import { RootState } from "./store";

const selectors = {
  getUserID: (state: RootState) => state.userAuth.userID,
  getName: (state: RootState) => state.userAuth.name,
  getEmail: (state: RootState) => state.userAuth.email,
  getPhone: (state: RootState) => state.userAuth.phone,
  getUserAuthJWT: (state: RootState) => state.userAuth.jwt,
  getIsLoggedIn: (state: RootState) => state.device.isLoggedIn,
  getIsLoading: (state: RootState) => state.device.isLoading,
  getWishList: (state: RootState) => state.device.wishList,
};

export { selectors };
