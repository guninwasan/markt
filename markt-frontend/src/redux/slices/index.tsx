export * from "./device-slice";
export * from "./user-auth-slice";
