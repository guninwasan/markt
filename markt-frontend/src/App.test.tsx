import { render, screen } from "@testing-library/react";
import App from "./App";
import { store } from "./redux";
import { Provider } from "react-redux";

const marktHomePageText = "Markt Home Page";
jest.mock("./pages/home-page", () => ({
  HomePage: () => <div>{marktHomePageText}</div>,
}));

describe("App", () => {
  it("should render the app properly and then home page", () => {
    render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    const homePage = screen.getByText(marktHomePageText);
    expect(homePage).toBeInTheDocument();
  });

  it("should match the snapshot", () => {
    const { container } = render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
