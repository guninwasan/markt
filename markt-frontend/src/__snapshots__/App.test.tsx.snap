// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div>
  <div>
    Markt Home Page
  </div>
  <footer
    class="sc-kvmDxl iAva"
  >
    <p
      class="sc-ZVgAE cEyIYj"
    >
      MARKT by 
      <i>
        Mizzica
      </i>
    </p>
    <nav
      aria-label="Footer navigation"
    >
      <div
        class="sc-dspCDH eFjVDu"
      >
        <a
          class="sc-fmtPfW hIZkS"
          href="/about"
        >
          About
        </a>
        <a
          class="sc-fmtPfW hIZkS"
          href="/privacy-policy"
        >
          Privacy Policy
        </a>
        <a
          class="sc-fmtPfW hIZkS"
          href="/terms-of-use"
        >
          Terms of Use
        </a>
        <a
          class="sc-fmtPfW hIZkS"
          href="/support"
        >
          Support / Report an issue
        </a>
      </div>
    </nav>
    <p
      class="sc-kUiVwB iCkLIl"
    >
      Â© 2024 All rights reserved.
    </p>
  </footer>
</div>
`;
